packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -39,8 +41,23 @@ class AndroidStudio implements Comparable<AndroidStudio> {
   }
 
   factory AndroidStudio.fromMacOSBundle(String bundlePath) {
-    final String studioPath = fs.path.join(bundlePath, 'Contents');
-    final String plistFile = fs.path.join(studioPath, 'Info.plist');
+    String studioPath = fs.path.join(bundlePath, 'Contents');
+    String plistFile = fs.path.join(studioPath, 'Info.plist');
+    String plistValue = iosWorkflow.getPlistValueFromFile(
+      plistFile,
+      null,
+    );
+
+    final String jetBrainsToolboxAppBundlePath = _jetBrainsToolboxAppMatcher.stringMatch(plistValue)?.split('=')?.last?.trim()?.replaceAll('"', '');
```

Comment:
Instead of writing this extract twice, make a method with a self-documenting name.